contrib/release_www.sh

```diff
@@ -34,29 +34,71 @@ if [ -z "$ELECTRUM_SIGNING_WALLET" ] || [ -z "$ELECTRUM_SIGNING_ADDRESS" ]; then
-info "VERSION: $VERSION"
+# get previous versions from WWW_DIR
+VERSION_STABLE=$(jq -r '.version' "$WWW_DIR/version")
+PREVIOUS_STABLE_VERSION=$VERSION_STABLE
+# returns version 0.0 if previous alpha or beta is not found
+VERSION_ALPHA=$(jq -r '.extradata.version_alpha // "0.0"' "$WWW_DIR/version")
+VERSION_BETA=$(jq -r '.extradata.version_beta // "0.0"' "$WWW_DIR/version")
+if [ "$VERSION_STABLE" == null ]; then
+    echo "Couldn't find previous version in $WWW_DIR/version"
+    exit 1
+fi
 
-ANDROID_VERSIONCODE_NULLARCH=$("$CONTRIB"/android/get_apk_versioncode.py "null")
+# get current electrum version from electrum/version.py
+RELEASE_VERSION=$("$CONTRIB"/print_electrum_version.py)
+if [[ "$RELEASE_VERSION" =~ [a] ]]; then
+    VERSION_ALPHA=$RELEASE_VERSION
+elif [[ "$RELEASE_VERSION" =~ [b] ]]; then
+    VERSION_BETA=$RELEASE_VERSION
+else
+    VERSION_STABLE=$RELEASE_VERSION
+fi
+info "RELEASE_VERSION: $RELEASE_VERSION"
+
+# keep the android versioncode for the stable release version or generate new if RELEASE_VERSION is stable
+if [[ "$VERSION_STABLE" != "$PREVIOUS_STABLE_VERSION" ]]; then
+    ANDROID_VERSIONCODE_NULLARCH=$("$CONTRIB"/android/get_apk_versioncode.py "null")
+    info "ANDROID_VERSIONCODE_NULLARCH: $ANDROID_VERSIONCODE_NULLARCH"
+else
+    ANDROID_VERSIONCODE_NULLARCH=$(jq -r '.extradata.android_versioncode_nullarch' "$WWW_DIR/version")
+    info "reusing old android versioncode for unstable release: $ANDROID_VERSIONCODE_NULLARCH"
+fi
 # ^ note: should parse as an integer in the final json
-info "ANDROID_VERSIONCODE_NULLARCH: $ANDROID_VERSIONCODE_NULLARCH"
+
+if [ "$ANDROID_VERSIONCODE_NULLARCH" == null ]; then
+    echo "No valid ANDROID_VERSIONCODE_NULLARCH: $ANDROID_VERSIONCODE_NULLARCH"
+    exit 1
+fi
 
 set -x
 
-info "updating www repo"
+info "updating www repo in $WWW_DIR"
 ./contrib/make_download "$WWW_DIR"
-info "signing the version announcement file"
-sig=$(./run_electrum -o signmessage $ELECTRUM_SIGNING_ADDRESS $VERSION -w $ELECTRUM_SIGNING_WALLET)
-# note: the contents of "extradata" are currently not signed. We could add another field, extradata_sigs,
-#       containing signature(s) for "extradata". extradata, being json, would have to be canonically
-#       serialized before signing.
+
+info "signing the stable version field"
+version_sig=$(./run_electrum -o signmessage "$ELECTRUM_SIGNING_ADDRESS" "$VERSION_STABLE" -w "$ELECTRUM_SIGNING_WALLET")
+
+# serialize extradata msg using the same method as in update_checker.py to prevent differences
+extradata=$(python3 -c "import json, sys; print(json.dumps({
+    'android_versioncode_nullarch': int(sys.argv[1]),
+    'version_alpha': sys.argv[2],
+    'version_beta': sys.argv[3]
+}, sort_keys=True, separators=(',', ':')))" "$ANDROID_VERSIONCODE_NULLARCH" "$VERSION_ALPHA" "$VERSION_BETA")
+
+# signing the hash of extradata instead of passing extradata directly,
+# so it doesn't get parsed as dict when passing it into signmessage,
```

Comment:
I started looking into this but it is turning into a quite rabbit hole...
The root cause is here:
https://github.com/spesmilo/electrum/blob/9b243169151ed18a06c3c02bebc508207d4b0456/run_electrum#L247-L249

I would say it's a bug that e.g. `$ ./run_electrum -v --testnet -w ~/.electrum/testnet/wallets/9dk -o signmessage tb1qq2tmmcngng78nllq2pvrkchcdukemtj5s6l0zu "5"` does not work, due to the message getting json-decoded to an int.
Anyway, I am still looking into this.